website/static/js/quickProjectSearchPlugin.js

```diff
@@ -0,0 +1,551 @@
+/**
+ * UI and function to quick search projects
+ */
+
+var $ = require('jquery');
+var m = require('mithril');
+var $osf = require('js/osfHelpers');
+
+// CSS
+require('css/quick-project-search-plugin.css');
+
+// XHR config for apiserver connection
+var xhrconfig = function(xhr) {
+    xhr.withCredentials = true;
+};
+
+
+var quickSearchProject = {
+    controller: function() {
+        var self = this;
+        self.nodes = m.prop([]); // Pending nodes waiting to be displayed
+        self.displayedNodes = m.prop([]); // Nodes that are rendered
+        self.nonMatchingNodes = m.prop([]); //Nodes that don't match search query
+        self.sortState = m.prop('dateDesc');
+        self.countDisplayed = m.prop();
+        self.next = m.prop();
+        self.loadingComplete = m.prop(false);
+        self.contributorMapping = {};
+        self.filter = m.prop();
+        self.totalLoaded = m.prop();
+
+        // Load first ten nodes
+        var url = $osf.apiV2Url('users/me/nodes/', { query : { 'embed': 'contributors'}});
+        var promise = m.request({method: 'GET', url : url, config : xhrconfig});
+        promise.then(function(result) {
+            self.countDisplayed(result.data.length);
+            result.data.forEach(function (node) {
+                self.nodes().push(node);
+                self.retrieveContributors(node)
+            });
+            self.next(result.links.next);
+            self.displayedNodes(self.nodes().splice(0, 10));
+            self.totalLoaded(self.displayedNodes().length)
+        });
+        promise.then(
+            function(){
+                if (self.next()) {
+                    self.recursiveNodes(self.next())
+                }
+                else {
+                    self.loadingComplete(true);
+                    m.redraw()
+                }
+            }
+        );
+
+        // Recursively fetches remaining user's nodes
+        self.recursiveNodes = function (url) {
+            if (self.nodes().length > 0) {
+                m.redraw()
+            }
+             var nextPromise = m.request({method: 'GET', url : url, config : xhrconfig, background : true});
+                nextPromise.then(function(result){
+                    result.data.forEach(function(node){
+                        self.nodes().push(node);
+                        self.retrieveContributors(node)
+                    });
+                    self.next(result.links.next);
+                    console.log(self.next());
+                    if (self.next()) {
+                        self.recursiveNodes(self.next())
+                    }
+                    else {
+                        self.loadingComplete(true);
+                        m.redraw()
+                    }
```

Comment:
There seems to be some repetition between 61-76 and 35-55. Is there a reason one couldn't use the same code as a subfunction?